src/lib/UIKitConfigProvider/types.ts

```diff
@@ -20,7 +22,7 @@ interface GroupChannelConfig {
         enableVideo: boolean;
       };
     };
-    replyType: 'none' | 'quote_reply' | 'thread';
+    replyType: CaseInsensitive<'none' | 'quote_reply' | 'thread'>;
```

Comment:
Question, why is this needed?

Reply:
Ahh we already have replyType which has all capital letters inside. https://github.com/sendbird/sendbird-uikit-react/blob/main/src/types.d.ts#L9. So I wanted to make it accept lowercase and capital letters both. 

I should've left comment about it. Just updated https://github.com/sendbird/sendbird-uikit-react/pull/561/commits/27596f85bbae94a78a3facff2cb9672d9e969400

Reply:
Ah~ I understand your point.

But one opinion is that it is preferable to maintain the original form of the API response as much as possible. Changing the snake_case may not be a significant issue, but modifying the type of values can lead developers further away from their initial expectations and impose mandatory tracking of history.

Moreover, when using the UI for rendering, it seems more beneficial in terms of maintainability and readability to map the values only once when passing them as props to UIKitConfigProvider, rather than performing checks for the two types every time.

```ts
if (config.replyType === 'thread' || config.replyType === 'THREAD') { ... } // x Component count

// vs

config={{ replyType: mapToReplyType('THREAD') }} 
if (config.replyType === 'thread') { ... } // x Component count
```

In practice, since only 'THREAD' is used in react-uikit, it is possible to perform the if statement check only for 'THREAD'. However, this can impose the need for history tracking on the person reading the code once again.

Reply:
@bang9 Yeah I see your point. 
I removed the custom CaseInsensitive type from UIKitConfigurationProvider. Instead, I convert the replyType config value from where it's used. 
https://github.com/sendbird/sendbird-uikit-react/pull/561/commits/98890740b5f158f2643b8c8a21c076c6c62727fd 

Instead of creating a separate util like mapToRelyType, just used toLowerCase(). But feel free to tell me if you see concern :)